src/sidebar/util/test/group-organizations-test.js

```diff
@@ -0,0 +1,174 @@
+    });
+
+    it ('should sort organizations by name', function () {
+      const org1 = orgFixtures.organization({ name: 'zzzzz' });
+      const org2 = orgFixtures.organization({ name: 'aaaaa' });
+      const org3 = orgFixtures.organization({ name: 'yyyyy' });
+      const groups = [orgFixtures.expandedGroup({ organization: org1 }),
+                      orgFixtures.expandedGroup({ organization: org2 }),
+                      orgFixtures.expandedGroup({ organization: org3 })];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      assert.equal(sortedGroups[0].organization.name, 'aaaaa');
+      assert.equal(sortedGroups[1].organization.name, 'yyyyy');
+      assert.equal(sortedGroups[2].organization.name, 'zzzzz');
+    });
+
+    it ('should sort organizations secondarily by id', function () {
+      const org1 = orgFixtures.organization({ name: 'zzzzz', id: 'zzzzz' });
+      const org2 = orgFixtures.organization({ name: 'zzzzz', id: 'aaaaa' });
+      const org3 = orgFixtures.organization({ name: 'zzzzz', id: 'yyyyy' });
+      const groups = [orgFixtures.expandedGroup({ organization: org1 }),
+                      orgFixtures.expandedGroup({ organization: org2 }),
+                      orgFixtures.expandedGroup({ organization: org3 })];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      assert.equal(sortedGroups[0].organization.id, 'aaaaa');
+      assert.equal(sortedGroups[1].organization.id, 'yyyyy');
+      assert.equal(sortedGroups[2].organization.id, 'zzzzz');
+    });
+
+    it ('should only include logo for first group in each organization', function () {
+      const org = orgFixtures.organization({ name: 'Aluminum' });
+      const org2 = orgFixtures.organization({ name: 'Zirconium' });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org2},
+        { name: 'Butane', organization: org2 },
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      assert(typeof sortedGroups[0].logo === 'string');
+      assert(typeof sortedGroups[1].logo === 'undefined');
+      assert(typeof sortedGroups[2].logo === 'string');
+      assert(typeof sortedGroups[3].logo === 'undefined');
+
+    });
+  });
+
+  context('when encountering missing data', function () {
+
+    it('should be able to sort without any groups in the default org', function () {
+      const org = orgFixtures.organization({ name:'Aluminum' });
+      const groups = [
+        { name: 'Aluminum', organization: org },
+        { name: 'Beryllium', organization: org},
+        { name: 'Cadmium', organization: org},
+      ];
+
+      const sortedGroups = groupsByOrganization(groups);
+
+      sortedGroups.forEach((group, idx) => {
+        assert.equal(group.name, groups[idx].name);
+      });
+
+    });
+
+    it('should omit any groups without an organization', function () {
```

Comment:
Since organizations are now mandatory for groups, do we still need this?

Reply:
I'm thinking long-term, as I'm not totally convinced that relationship is going to stay that way. But I can pull it if you like!

Reply:
My general preference is to take advantage of assumptions that can be made to simplify the code where possible. We can always change the code later. That will be the easy part compared to making an API change.